Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs

```diff
@@ -20,19 +20,29 @@ public ScheduleRepository(IOptions<MongoSettings> options)
         public Task<List<Schedule>> GetByUserPrimarySkillAsync(IEnumerable<Guid> userIds, DateTime date, Guid primarySkillId)
         {
             var filter = _filterBuilder.In(u => u.UserId, userIds) & _filterBuilder.Eq(u => u.UserPrimarySkill.Id, primarySkillId);
+            return GetFindFluentByDate(filter, date, 1).ToListAsync();
+        }
+
+        public Task<Schedule> GetByDatePeriodAsync(Guid userId, DateTime date, int daysNum)
+        {
+            var filter = _filterBuilder.Eq(u => u.UserId, userId);
+            return GetFindFluentByDate(filter, date, daysNum).FirstOrDefaultAsync();
+        }
+
+        private IFindFluent<Schedule, Schedule> GetFindFluentByDate(FilterDefinition<Schedule> filter, DateTime date, int daysNum)
```

Comment:
may be write daysNum=1 ? so you don't need pass it from 23, it's just an option to try